polar_route/mesh_generation/mesh_builder.py

```diff
@@ -204,6 +205,51 @@ def initialize_cellboxes(self, bounds, cell_width, cell_height):
                     cellbox = CellBox(cell_bounds, cell_id)
                 cellboxes.append(cellbox)
         return cellboxes
+    
```

Comment:
The reason Sonar is complaining about pep8 here is because the parameter 'Dataloader' is capitalised. Pep8 dictates that any variable should start with a lowercase letter, and classes with an uppercase.  I'm a little confused with your comment about 'Dataloader' being a class that is passed as an argument - We don't actually have a class 'Dataloader' in PolarRoute, we have an interface, factory, and various implementations, but the class 'Dataloader' doesn't exist. 

I expect the function to follow the form:
```
def add_dataloader(self, dataloader: DataLoaderInterface):
        
        # add dataloader to the list of dataloaders....
```

Where a single variable of type DataLoaderInterface (_and hence any object inheriting from the interface_) is passed. This also removed the need for the additional variables params, bounds, name etc to the function as they are already attributes of the passed dataloader.

Reply:
Happy happy with whatever works really - I don't want to enforce my coding-style on anyone else if you're following standard conventions as well. 

I'd assume the dataloader object has been initialised before being passed to the 'add_dataloader()' function. Something like 

```
from polar_route.dataloaders.factory import DataLoaderFactory
from polar_route.mesh_generation.boundary import Boundary
from polar_route.mesh_generation.mesh_builder import MeshBuilder

config = {...}
params = {...}

mesh_builder = MeshBuilder(config)

dl_factory = DataLoaderFactory()
bounds = Boundary(0, 0, 0, 0)
# getting a DL from the factory, but this can come from anywhere
amsr = dl_factory.get_dataloader('amsr2', bounds, params)

# the function is passed an initialised dataloader, containing all the info the mesh_builder needs
mesh_builder.add_dataloader(amsr)


mesh = mesh_builder.build_environmental_mesh()
```